zio-http/src/main/scala/zio/http/Request.scala

```diff
@@ -1,199 +1,55 @@
 package zio.http
 
-import io.netty.handler.codec.http.{DefaultFullHttpRequest, FullHttpRequest, HttpRequest}
-import zio.Unsafe
 import zio.http.model._
 import zio.http.model.headers._
-import zio.http.netty._
-import zio.http.service.Ctx
 
-import java.io.IOException
+import java.net.InetAddress
 
-trait Request extends HeaderExtension[Request] { self =>
-
-  /**
-   * Returns a string representation of the request, useful for debugging,
-   * logging or other purposes. It contains the essential properties of HTTP
-   * request: protocol version, method, URL, headers and remote address.
-   */
-  final override def toString = s"Request($version, $method, $url, $headers)"
-
-  /**
-   * Updates the headers using the provided function
-   */
-  final override def updateHeaders(update: Headers => Headers): Request = self.copy(headers = update(self.headers))
+final case class Request(
```

Comment:
This change will need to be benchmarked since one of the reasons `Request` was a trait was to provide the ability to virtualize some of its fields, rather than be forced to materialize the body, header, method, url, version, and remote address (including its decoding and allocation cost).

Now, I am not sure if this was actually utilized anymore, or if the `trait` was more "forward thinking". We need to benchmark to make sure no regression here.

To accomodate high-performance use-cases, if necessary, we could introduce a `trait RequestLike` that defines all the same fields, and then read-only use cases could require `RequestLike` rather than `Request`.

Reply:
I assume your comment further down allays some the concern here?  Still, I will be running benchmarks.